sklearn/decomposition/tests/test_incremental_pca.py

```diff
@@ -228,7 +229,7 @@ def test_incremental_pca_batch_signs():
         ipca = IncrementalPCA(n_components=None, batch_size=batch_size).fit(X)
         all_components.append(ipca.components_)
 
-    for i, j in zip(all_components[:-1], all_components[1:]):
+    for i, j in itertools.pairwise(all_components):
```

Comment:
Note for other reviewers: this is an automated replacement, `ruff` is doing this when using target-version "py310"